tests/spec/fs.link.spec.js

```diff
@@ -111,3 +111,121 @@ describe('fs.link', function() {
     });
   });
 });
+
+
+
+
+
+describe('fs.promises.link', function() {
+  beforeEach(util.setup);
+  afterEach(util.cleanup);
+
+  it('should be a function', function() {
+    var fs = util.fs();
+    expect(fs.promises.link).to.be.a('function');
+  });
+
+  it('should create a link to an existing file', function(done) {
+    var fs = util.fs();
+
+    fs.open('/myfile', 'w+', function(error, fd) {
+      if(error) throw error;
+
+      fs.close(fd, function(error) {
+        if(error) throw error;
+
+        fs.promises.link('/myfile', '/myotherfile').then(function(){
+
+          fs.stat('/myfile', function(error, result) {
+            if(error) throw error;
+      
+            var _oldstats = result;
+            fs.stat('/myotherfile', function(error, result) {
+              expect(error).not.to.exist;
+              expect(result.nlinks).to.equal(2);
+              expect(result.dev).to.equal(_oldstats.dev);
+              expect(result.node).to.equal(_oldstats.node);
+              expect(result.size).to.equal(_oldstats.size);
+              expect(result.type).to.equal(_oldstats.type);
+              done();
+            });
+          },
+          function(error){throw error;}
+          );
+        });
+      });
+    });
+  });
+
+  it('should create hard link to identical data node', function(done) {
+    var fs = util.fs();
+    var contents = 'Hello World!';
+
+    fs.writeFile('/file', contents, function(err) {
+      if(err) throw err;
+
+      fs.promises.link('/file', '/hlink', function(err) {
```

Comment:
This is mixing callbacks and promises.  You shouldn't be passing a callback to link here.